packages/11ty/_includes/web-components/image-sequence/index.js

```diff
@@ -311,55 +356,81 @@ class ImageSequence extends LitElement {
    * 
    **/ 
   willUpdate(changedProperties) {
+    // Determine the animation indices, preload them, and then do the rotation 
     if (changedProperties.has('rotateToIndex') && this.rotateToIndex!==false) {
-      this.performRotation(this.rotateToIndex)
+      const frameCount = this.rotateToIndex - this.index
+      const animationIndices = Array(frameCount).fill(0).map( (_,i) => this.index + i + 1 )
+      this.#preloadImages( animationIndices ).then( this.performRotation(this.rotateToIndex) )
     }
 
+    // Draws `animationIndex` directly to canvas (for use )
+    if (changedProperties.has('animationIndex')) {
+      if (!this.animationIndex) { return }
+      if (this.images[this.animationIndex] === null) { return }
+
+      this.#paintCanvas(this.images[this.animationIndex])
+    } 
+
     if (changedProperties.has('index') && this.someImagesLoaded) {
-      this.#preloadImages()
 
       if (this.bufferReady) {
-        this.#paintCanvas()      
+        this.#paintCanvas()
+        return      
       }
+
+      this.#preloadImages().then( () => {
+        this.performRotation(this.index)
+      })
+
     }
 
+    // Load enough to prepare for interaction
     if (changedProperties.has('visible') && !this.visible) {
-      this.#preloadImages()
+      this.#preloadImages().then( () => this.#paintCanvas() )
     }
   }
 
   /**
    * @function #preloadImages
    * 
-   * Loads the k images behind and ahead of this.index
+   * Loads images of this.indexesToBuffer -- ahead and behind `this.index`
+   * 
+   * @returns {Promise} - Promise resolution for all preloads
+   * 
    **/
-  #preloadImages() {
+  #preloadImages(indices) {
     if (!this.images.some(i => i === null)) { return }
 
-    // Really just making buffer counting ergonomic / readable here
-    // const indexesToBuf = Array(this.bufferSize).fill(0).map( (_,i) => ( this.images.length + this.index + i - Math.round(this.bufferSize/2) ) % this.images.length )
-    this.images.forEach( (image,i) => {
+    const indexesToLoad = indices ?? this.indexesToBuffer
+    const fetches = this.images.map( (image,i) => {
```

Comment:
```suggestion
    const image = this.images.map((image,i) => {
```
or
```
    const response = this.images.map((image,i) => {
```

Reply:
These changes are now incorporated (along with the switch of computing bufferWindow and bufferReady with private functions, and attendant variable renames, see https://github.com/thegetty/quire/blob/ec1a79fc19adc1be430c3ea79ae12914079f1928/packages/11ty/_includes/web-components/image-sequence/index.js#L404 )